src/rulesets/Base/sort.jl

```diff
@@ -33,3 +37,54 @@ function rrule(::typeof(sort), xs::AbstractVector; kwargs...)
     end
     return ys, sort_pullback
 end
+
+#####
+##### `sortslices`
+#####
+
+function rrule(::typeof(sortslices), x::AbstractArray{<:Number}; dims::Integer, kw...)
+    p = sortperm(collect(eachslice(x; dims=dims)); kw...)
+    inds = ntuple(d -> d == dims ? p : (:), ndims(x))
+    function sortslices_pullback(dy)
+        # No actual need to zero this, and if you didn't, then you could widen eltype
+        # Also, you could use similar(dy) here not x, same size?
+        dx = _zerolike_writeat(x, unthunk(dy), (), inds...)
```

Comment:
Do we need the `unthunk`?
If so, should we push it down inside the `_zerolike_writeat` ?

Reply:
I think that ideally, `_zerolike_writeat` should be upgraded to return an InplaceThunk. And eventually it should be called grad_getindex or something, too.

I'm not sure whether it should handle un-thunking. I guess it wouldn't hurt to add a method. But since most rules at present call unthunk explicitly, maybe it's clearer to call it here?

Reply:
The only other place it's called at present is:

https://github.com/JuliaDiff/ChainRules.jl/blob/main/src/rulesets/Base/array.jl#L364-L373

Reply:
Arguably we shouldn't be unthunking if the destination that we are writing into can accept `Any`.
(but practically that case doesn't really matter since performance is already shot.  And likely Zygote will hate that)

Reply:
Yes this is far from being a high-performance function!

If you don't take the shortcut above, then not all entries were unique, and thus `_zerolike_writeat` has to copy dy into dx at some nontrivial indices. So it has to slice up `dy`, I don't think it can write just one thunk anywhere.